plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/p3order/LayerSweepCrossingMinimizer.java

```diff
@@ -179,6 +180,15 @@ private void compareDifferentRandomizedLayouts(final GraphInfoHolder gData) {
         graphsWhoseNodeOrderChanged.clear();
 
         int bestCrossings = Integer.MAX_VALUE;
+        if (gData.lGraph().getProperty(LayeredOptions.PRESERVE_ORDER) != OrderingStrategy.NONE) {
+            // Set the current order of nodes and ports as the currently best order.
+            // If no unnecessary crossings are produced by this solution this order is preserved.
```

Comment:
Maybe I miss something but don't you have to sweep at least once? In its current form I read the code like this:

1) count the crossings of the current node order
2) set the best node order to the current order (which may already be the case?)
3) if no crossing exists, finish crossing minimization.

Reply:
The steps are:
0: (Before this) Order nodes and portss as in the model and produce (hopefully no unnecessary crossings)
1: Count crossings
2: set the current node order as the bestNode order (which is not the case)
3: if no crossings exist we have a special case in which we know the solution is optimal and the finish crossing minimization
4: If a better solution exists it will override the initial solution. Otherwise it is preserved.

Can you elaborate? Did I miss something about this algorithm?

Reply:
I'm referring to step 3). I thought the result of our discussion a week back was that even with no crossings a sweep could be used (and is required) to properly position dummy nodes of long edges etc.

Reply:
Yes. That was the initial idea. Now dummy nodes are included in the initial ordering. However, I think it might be good to use the preserved ordering for an initial sweep nonetheless. This might produce better results, if the initial ordering is conflicting.